### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by VickMellon**
Unit tests for ZkSigner, EthSigner and Wallet was added, also are mocks for Provider and EthProvider was generated.
This tests validate that main entities is properly created, then check that composing and signing of all types of Txs is as expected.

### CODE REVIEW

File: EthProvider.go [hunk 1]
```diff
@@ -12,6 +12,8 @@ import (
 	"math/big"
 )
 
+//go:generate mockery -name=EthProvider -output=./ -outpkg=zksync -filename=EthProviderMock_test.go -structname=EthProviderMock -inpkg
```
Comment (StanislavBreadless) [inline EthProvider.go:15]: is it needed?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	test -f ./libs/libzks-crypto.so || curl -L https://github.com/zksync-sdk/zksync-crypto-c/releases/download/v0.1.1/zks-crypto-linux-x64.so --output ./libs/libzks-crypto.so
+endif
+
+tests: download
+	CGO_LDFLAGS="-L./libs" go test  -race -v -count=1 .
+
+generate:
+	go install github.com/vektra/mockery/cmd/mockery
+	go install github.com/golang/mock/mockgen
+	go generate .
```
Comment (StanislavBreadless) [inline Makefile:17]: We should strive to keep newlines at the end of files

File: .gitignore [hunk 3]
```diff
@@ -1 +1,2 @@
-.idea
\ No newline at end of file
+.idea
+libs/
```
Comment (StanislavBreadless) [inline .gitignore:2]: there should be a newline at the end of file

**Review by StanislavBreadless — COMMENTED** [review]
Generally nice.

While mocking is good, I think that the tests should aim to interact with the real API. While it is not  a best practice to interact with an outside API in the CI tests, it is still the best way to check that the SDK functionality actually works when working with real server and not just the mocks. In our repo we use zkSync server bootstrapped inside the CI to test the SDK.

Bootstrapping your own zkSync server in the CI may not be feasible & the server may not be fully up-to-date, so I suggest that we use the rinkeby provider

**Comment by StanislavBreadless** [comment]
> * also can't swap the same tokens (ETH), but this Tx performing and expecting for this error

One of the wallets could deposit rinkeby USDC, while the other one deposits rinkeby ETH

**Comment by VickMellon** [comment]
Ok, USDC deposit for wallet 1 was added and Swap test completed

**Comment by VickMellon** [comment]
This is sample of successful tests PASS
```
victorkavunenko@~/go/src/github.com/zksync-sdk/zksync-go$ make integration-test
mkdir -p libs
test -f ./libs/libzks-crypto.a || curl -L https://github.com/zksync-sdk/zksync-crypto-c/releases/download/v0.1.1/zks-crypto-macos-x64.a --output ./libs/libzks-crypto.a
CGO_LDFLAGS="-L./libs" go test  -race -v -count=1 ./IntegrationTests
=== RUN   TestFullFlow
=== RUN   TestFullFlow/wallet_1
mnemonic - clay leaf metal cabbage trust cute cabin wolf pill object wrist carry super file hat
res pkh [237, 2, 222, 163, 121, 234, 164, 247, 59, 199, 167, 89, 99, 221, 61, 32, 142, 51, 99, 80]
ethSigner address - 0x3c941a8BB4f7FBB2C44C917e5536148841ead201
Waiting for fulfillment Eth Tx 0x2b23b0e95882e25625339b05e43501dbe30d7cf74ed6ca57e2ad79c6430944db.......DONE
Waiting for fulfillment USDC Eth Tx 0x5548316340a64f1fbd1cc4633430ace9b934b07c27535f56470de8f1bb85c76a.............DONE
Waiting for deposit Eth Tx 0xb26af667c66bc29babaade3e860d5ba06a65232d46ad46344b9efd3e44884e0d.......DONE
Waiting for new ZkSync account 0x3c941a8BB4f7FBB2C44C917e5536148841ead201............................DONE
Waiting for approve deposit USDC Eth Tx 0x3ad63f05da97650159142465523999764f77efaa18055382358427d9b3f9814d......DONE
Waiting for deposit USDC Eth Tx 0xa6d959e75b9080823faf13e262f006617983e45eeba5a90470f07973d05e3668.......DONE
Waiting for setAuthPubkeyHash Eth Tx 0xfaf873b9214b787433e715947becdfa4c1c5a0702f067e7b6f3c262a38da06ce......DONE
Waiting for ChangePubKeyOnchain ZkSync Tx sync-tx:9851b41063646d352743f3a24874f18b20355eca495c450cd9e9fa9c115b737c.DONE
=== RUN   TestFullFlow/wallet_2
mnemonic - deer asthma fold grit rug apology wife social jump frequent reward route need earn remind
res pkh [106, 205, 48, 175, 121, 166, 61, 28, 25, 61, 238, 173, 151, 6, 106, 226, 209, 148, 154, 16]
ethSigner address - 0xE20A10ADBBfA2021C5902F1A58248EF2bCfC3241
Waiting for fulfillment Eth Tx 0xc71e757df38c7f2bb59f295ad54eb5ab389c9b1a332fcdf8ec59804330ce9dc3.....DONE
Waiting for deposit Eth Tx 0xc94822bd93539d3dd24fc0a22d957b146220fca78b8180b38901cecd54d21cf0......DONE
Waiting for new ZkSync account 0xE20A10ADBBfA2021C5902F1A58248EF2bCfC3241............................DONE
Waiting for ChangePubKeyECDSA ZkSync Tx sync-tx:7401e8ba9ef53a9fbee18788d79efcf9cd23a3932d3697af83c4b24d861230ca.DONE
=== RUN   TestFullFlow/transfer
Waiting for SyncTransfer ZkSync Tx sync-tx:514253a8b73d598f0192de83ef2ca0b5038f6e24b3c8f3710d2c7d4d36100b08.DONE
=== RUN   TestFullFlow/withdraw
Waiting for SyncWithdraw ZkSync Tx sync-tx:083abbd23a2a7c3988af10718858791affd225911bc8787fd0c1b70063c4db85.DONE
=== RUN   TestFullFlow/mint_NFT_1
Waiting for SyncMintNFT ZkSync Tx sync-tx:87bc487c14dc9abe8c4c5806286a825b3cabe6144555e044ded50da10f715e8f.DONE
=== RUN   TestFullFlow/mint_NFT_2
Waiting for SyncMintNFT ZkSync Tx sync-tx:402a0c54859f4724fedd1c5d0401cea92fdb4d0925276928428f39a8225c91ff.DONE
=== RUN   TestFullFlow/full_exit_NFT_1
Waiting for FullExitNFT Eth Tx 0x273d9ccfe6ca57ec96b935471121fa9796b0cf00ba39aa55fc9de43a7c3829d7......DONE
=== RUN   TestFullFlow/withdraw_NFT_2
Waiting for SyncWithdrawNFT ZkSync Tx sync-tx:4c5025df6ba3e40cfb4edca88584159fba49a155658ca3290f5e525fd99406cf.DONE
=== RUN   TestFullFlow/swap
Waiting for SyncSwap ZkSync Tx sync-tx:e60ae6a308daf6aca4cf662f34c5939314bddaa66144d026bf9d7deb9ff0a171.DONE
=== RUN   TestFullFlow/full_exit
Waiting for FullExit Eth Tx 0x0fac3f75cfb3366fffed7e526b1ea14eac3955c90c02d853d1a8c2c46b8fa05c....DONE
--- PASS: TestFullFlow (301.51s)
    --- PASS: TestFullFlow/wallet_1 (169.39s)
    --- PASS: TestFullFlow/wallet_2 (90.34s)
    --- PASS: TestFullFlow/transfer (3.32s)
    --- PASS: TestFullFlow/withdraw (3.11s)
    --- PASS: TestFullFlow/mint_NFT_1 (3.03s)
    --- PASS: TestFullFlow/mint_NFT_2 (3.02s)
    --- PASS: TestFullFlow/full_exit_NFT_1 (13.73s)
    --- PASS: TestFullFlow/withdraw_NFT_2 (3.09s)
    --- PASS: TestFullFlow/swap (3.42s)
    --- PASS: TestFullFlow/full_exit (9.07s)
PASS
ok  	github.com/zksync-sdk/zksync-go/IntegrationTests	301.675s
```

**Comment by StanislavBreadless** [comment]
@VickMellon The tests should be run in GitHub Actions CI

**Review by StanislavBreadless — APPROVED** [review]

**Pull request merged**